test/extended/baremetal/common.go

```diff
@@ -58,9 +58,25 @@ func skipIfUnsupportedPlatformOrConfig(oc *exutil.CLI, dc dynamic.Interface) {
 	}
 }
 
+func skipIfProvisioningNetworkSet(dc dynamic.Interface) {
+	g.By("checking Provisioning Network CIDR")
+
+	provisioningClient := provisioningClient(dc)
+	provisioningConfig, err := provisioningClient.Get(context.Background(), "provisioning-configuration", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	provisioningSpec, _, err := unstructured.NestedMap(provisioningConfig.UnstructuredContent(), "spec")
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	_, provisioningNetworkCIDRfound, err := unstructured.NestedString(provisioningSpec, "provisioningNetworkCIDR")
```

Comment:
Instead getting `provisioningSpec` first and after `provisioningNetworkCIDR`, maybe we can do this in one step like in here https://github.com/openshift/origin/blob/d335c644c13b920b3bc9ab404133788ebd76856a/test/extended/baremetal/helper.go#L184